webview-ui/src/Periscope/Periscope.tsx

```diff
@@ -4,39 +4,44 @@ import { InitialState } from "../../../src/webview-contract/webviewDefinitions/p
 import { ErrorView } from "./ErrorView";
 import { NoDiagnosticSettingView } from "./NoDiagnosticSettingView";
 import { SuccessView } from "./SuccessView";
-import { getStateManagement } from "../utilities/state";
+import { useStateManagement } from "../utilities/state";
 import { stateUpdater, vscode } from "./state";
 
 export function Periscope(initialState: InitialState) {
-    const {state, eventHandlers, vsCodeMessageHandlers} = getStateManagement(stateUpdater, initialState);
+    const { state, eventHandlers } = useStateManagement(stateUpdater, initialState, vscode);
 
     useEffect(() => {
-        vscode.subscribeToMessages(vsCodeMessageHandlers);
         sendUploadStatusRequest();
-    }, []); // Empty list of dependencies to run only once: https://react.dev/reference/react/useEffect#useeffect
+    });
```

Comment:
This removal of the empty list of dependencies (done because there were hidden dependencies) will result in more frequent upload status requests. It's worth testing this and seeing if it's too chatty.

Reply:
Manually tested, and not noticed any different behaviour.